shadowsocks/tcp.go

```diff
@@ -189,6 +191,7 @@ func (s *tcpService) Start() {
 			}()
 			connStart := time.Now()
 			clientConn.(*net.TCPConn).SetKeepAlive(true)
+			clientConn.SetReadDeadline(connStart.Add(s.readTimeout))
```

Comment:
Add a comment like "Set a deadline for connection authentication"

Reply:
done